src/sinks/util/batch.rs

```diff
@@ -85,20 +86,10 @@ pub struct Merged;
 pub struct Unmerged;
 
 /// Event batching behavior.
-// TODO: We should/could probably derive the impl of `Configurable` such that it uses `D` to get the
-// default batch settings, since we'll need an effective way to encode different defaults for the
-// various sinks. Sort of middleground between "define defaults per-field" and "this type has its
-// own default" but I don't think it will end up being too messy.
-//
-// Differently, maybe we just write a custom `Default` implementation for `BatchConfig` such that it
-// extracts the actual default values from `D`? What gets tricky there is there is that users would
-// need to specify `#[serde(default)]` for that default value to get pulled up correctly, whereas
-// the custom `Configurable` implementation does that automatically for us.
-//
-// Thinking even _more_ about it, maybe we could actually just define a per-field default here that
-// derives from the consts in `D`? I think _that_ might work... but we don't pull up per-field
-// defaults to the callsite of whatever is using `BatchConfig`, IIRC, so we wouldn't actually show
-// the default value at the callsite? Hmph.
```

Comment:
I'll take another look at this to see if we can put the configurable example macro on the impl of SinlBatchSettings instead, so it is more clear. Not sure it will work though.

Reply:
Well, I've concluded that there isn't a way to do this as it stands, but that it might be possible. I think a consult with @tobz would help determine the realisticness and effort level and worth it factor.

As it stands, I think this solution is not so bad, because the sinks using `BatchConfig` are already using configurable(derived) , and the `SinkBatchSettings` don't have member vars that devs might unknowingly add metadata too.